### PR TITLE
feat: New layer legend, scrolling and mobile styling AB#106191 AB#106004 AB#106016

### PR DESCRIPTION
**Pull request by ChrisSkj**
### What's new?

- Add a simple legend layout for leakage detection.
- Make the legend content scrollable and style the scrollbar to look shiny
- Adjust the legend to be more mobile friendly while we wait for the new mobile design
- Never a PR without bumping us sum CLI

2023-03-30 14_51_45-Echo _ Development

2023-03-30 14_51_24-Echo _ Development

### CODE REVIEW

File: src/layers/external/leakage/SubseaLeakageLayerConfiguration.ts [hunk 1]
```diff
@@ -105,6 +106,10 @@ export class SubseaLeakageLayerConfiguration implements LayerConfiguration {
         return SubseaLeakageTooltip;
     }
 
+    public getLegendComponent() {
+        return SubseaLeakageLegend;
+    }
+
```
Comment (Tobzor) [inline src/layers/external/leakage/SubseaLeakageLayerConfiguration.ts:112]: This has been moved to the uiElements instance on layers

File: src/layers/mapHub/marineInfrastructure/legend/MarineInfrastructureLegend.tsx [hunk 2]
```diff
@@ -53,7 +52,7 @@ interface InstallationProps {
 
 function MarineInfrastructureLayerLegend({ subLayer }: InstallationProps) {
     const [isOpen, setIsOpen] = useState(false);
-    const isAddedToMap = useObservable(subLayer.isAddedToMap);
+    const isMobile = useIsMobile();
```
Comment (Tobzor) [inline src/layers/mapHub/marineInfrastructure/legend/MarineInfrastructureLegend.tsx:55]: useIsMobile is the old hook version thingy -- we use Erlends new deviceInfo other places now

File: src/layers/mapHub/marineInfrastructure/legend/MarineInfrastructureLegend.tsx [hunk 3]
```diff
@@ -155,11 +159,17 @@ function MarineInfrastructureSubLayerLegend(
     const toggleOpen = () => {
         isOpen ? setIsOpen(false) : setIsOpen(true);
     };
+    const isMobile = useIsMobile();
```
Comment (Tobzor) [inline src/layers/mapHub/marineInfrastructure/legend/MarineInfrastructureLegend.tsx:162]: deviceInfo from Erlend -- not sure if it has been moved to utils yet or if its still in framework

File: src/layers/mapHub/marineInfrastructure/legend/MarineInfrastructureLegend.tsx (hunk 2 above)
Comment (ChrisSkj) [reply src/layers/mapHub/marineInfrastructure/legend/MarineInfrastructureLegend.tsx:55]: I'll get back to this one. Will do a big cleanup US for all isMobile instances

File: src/layers/mapHub/marineInfrastructure/legend/MarineInfrastructureLegend.tsx (hunk 3 above)
Comment (ChrisSkj) [reply src/layers/mapHub/marineInfrastructure/legend/MarineInfrastructureLegend.tsx:162]: I'll get back to this one. Will do a big cleanup US for all isMobile instances

**Review by Tobzor — APPROVED** [review]

**Pull request merged**